aws-lc-rs/src/cipher.rs

```diff
@@ -737,8 +818,79 @@ fn decrypt_aes_cbc_mode<'in_out>(
     Ok(in_out)
 }
 
+fn encrypt_aes_cfb_mode(
+    key: &SymmetricCipherKey,
+    mode: OperatingMode,
+    context: EncryptionContext,
+    in_out: &mut [u8],
+) -> Result<DecryptionContext, Unspecified> {
+    #[allow(clippy::match_wildcard_for_single_variants)]
+    let key = match &key {
+        SymmetricCipherKey::Aes128 { enc_key, .. } | SymmetricCipherKey::Aes256 { enc_key, .. } => {
+            enc_key
+        }
+        _ => return Err(Unspecified),
+    };
+
+    let mut iv = {
+        let mut iv = [0u8; AES_CFB_IV_LEN];
+        iv.copy_from_slice((&context).try_into()?);
+        iv
+    };
+
+    let cfb_encrypt: fn(&AES_KEY, &mut [u8], &mut [u8]) = match mode {
+        // TODO: Hopefully support CFB1, and CFB8
+        OperatingMode::CFB128 => aes_cfb128_encrypt,
+        _ => {
+            // this indicates a programming error and shouldn't happen
+            return Err(Unspecified);
+        }
+    };
+
+    cfb_encrypt(key, &mut iv, in_out);
+    iv.zeroize();
+
+    Ok(context.into())
+}
+
+fn decrypt_aes_cfb_mode<'in_out>(
+    key: &SymmetricCipherKey,
+    mode: OperatingMode,
+    context: DecryptionContext,
+    in_out: &'in_out mut [u8],
+) -> Result<&'in_out mut [u8], Unspecified> {
+    #[allow(clippy::match_wildcard_for_single_variants)]
+    let key = match &key {
+        SymmetricCipherKey::Aes128 { enc_key, .. } | SymmetricCipherKey::Aes256 { enc_key, .. } => {
+            enc_key
+        }
+        _ => return Err(Unspecified),
+    };
+
+    let mut iv = {
+        let mut iv = [0u8; AES_CFB_IV_LEN];
+        iv.copy_from_slice((&context).try_into()?);
+        iv
+    };
+
+    let cfb_decrypt: fn(&AES_KEY, &mut [u8], &mut [u8]) = match mode {
+        // TODO: Hopefully support CFB1, and CFB8
+        OperatingMode::CFB128 => aes_cfb128_decrypt,
+        _ => {
+            // this indicates a programming error and shouldn't happen
+            return Err(Unspecified);
+        }
+    };
+
+    cfb_decrypt(key, &mut iv, in_out);
+
+    iv.zeroize();
```

Comment:
NP: The IV is not secret, so no need to zeroize (but I think we do likewise in many other places).

Reply:
It's a pattern followed elsewhere in the existing code, but agree it's technically not considered secret.